cpp/src/arrow/util/trie.h

```diff
@@ -125,6 +126,9 @@ class ARROW_EXPORT Trie {
   int32_t Find(util::string_view s) const {
     const Node* node = &nodes_[0];
     fast_index_type pos = 0;
+    if (s.length() > static_cast<size_t>(kMaxIndex)) {
```

Comment:
@pitrou One small difference from your original patch.  This changed from >= to >.  Could you review that?  My test of the boundary cases considered a string of length 2^n-1 as valid and so I either needed to remove that case or change the comparison (and I chose the latter).

Reply:
Nice, thank you!